csrc/fusion_segmenter.cpp

```diff
@@ -3548,15 +3549,28 @@ bool SegmentCandidateFinder::codeGenSupportedMerge(
   NVF_ERROR(
       areDirectlyConnected(group1, group2),
       "only support testing immediate producer-consumer groups");
-  auto h = tryMerge(segmented_fusion_.get(), runtime_info_, group1, group2);
+  if (options_.only_segment_resharding_exprs) {
+    for (auto group : {group1, group2}) {
+      for (auto expr : group->exprs()) {
+        if (isResharding(expr)) {
+          return false;
+        }
+      }
+    }
+    return true;
+  }
+  auto h = tryMerge(segmented_fusion_.get(), runtimeInfo(), group1, group2);
   return h.has_value();
 }
 
 // TODO: consider caching the heuristics value so tryMerge doesn't have to be
 //       called twice
 ScheduleHeuristic SegmentCandidateFinder::deriveHeuristic(
     SegmentedGroup* group) {
-  auto h = tryMerge(segmented_fusion_.get(), runtime_info_, group);
+  if (!runtime_info_.has_value()) {
```

Comment:
This should only happen when segmenting resharding expressions, right? I'd write:

```suggestion
  // We don't need to generate a heuristic for multidevice segments at this moment
  if (options_.only_segment_resharding_exprs) {
     return ScheduleHeuristic::None;
  }
```

This way, we wouldn't silently mask a bug with a null `runtime_info_`.

Reply:
Right, that's better as you suggest